webapp/javascript/pages/IntroPages/SignIn/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import cx from 'classnames';
+import Icon from '@ui/Icon';
+import { faGithub } from '@fortawesome/free-brands-svg-icons/faGithub';
+import InputField from '@ui/InputField';
+import StatusMessage from '@ui/StatusMessage';
+import { useAppDispatch, useAppSelector } from '@pyroscope/redux/hooks';
+import { logIn } from '@pyroscope/services/users';
+import {
+  loadCurrentUser,
+  selectCurrentUser,
+} from '@pyroscope/redux/reducers/user';
+import { GitlabIcon, GoogleIcon } from '../Icons';
+import Divider from '../Divider';
+import inputStyles from '../InputGroup.module.css';
+import styles from '../IntroPages.module.css';
+import buttonStyles from './buttons.module.css';
+
+function SignInPage() {
+  const history = useHistory();
+  const dispatch = useAppDispatch();
+  const currentUser = useAppSelector(selectCurrentUser);
+  const [form, setForm] = useState({
+    username: '',
+    password: '',
+    errors: [],
+  });
+
+  const handleFormChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name } = event.target;
+    const { value } = event.target;
+    setForm({ ...form, [name]: value });
+  };
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      const { username, password } = {
+        ...form,
+      };
+
+      const res = await logIn({ username, password });
+      if (res.isOk) {
+        dispatch(loadCurrentUser());
+        return;
+      }
+
+      throw res.error;
+    } catch (e: any) {
+      setForm({ ...form, errors: e.errors || [e.message] });
+    }
+  }
+
+  useEffect(() => {
+    if (currentUser) {
+      history.push('/');
+    }
+  }, [currentUser]);
+
+  return (
+    <div className={styles.loginWrapper}>
+      <form className={styles.form} onSubmit={handleSubmit}>
+        <div className={styles.formHeader}>
+          <div className={styles.logo} />
+          <h1>Welcome to Pyroscope</h1>
+          <h3>Log in to continue</h3>
+        </div>
+        <div>
```

Comment:
see my comment below about these fields being conditional. The regular sign in inputs are only available if you have internal authentication enabled, e.g:
```
auth:
  internal:
    enabled: true
```